ktor-features/ktor-auth/jvm/src/io/ktor/auth/OAuth1a.kt

```diff
@@ -162,18 +163,21 @@ private suspend fun requestOAuth1aAccessToken(
     val authHeader = createUpgradeRequestTokenHeaderInternal(consumerKey, token, nonce)
         .signInternal(HttpMethod.Post, baseUrl, secretKey, params)
 
-    val body = client.post<String>(baseUrl) {
-        header(HttpHeaders.Authorization, authHeader.render(HeaderValueEncoding.URI_ENCODE))
-        header(HttpHeaders.Accept, "*/*")
-        // some of really existing OAuth servers don't support other accept header values so keep it
-
-        body = WriterContent(
-            { params.formUrlEncodeTo(this) },
-            ContentType.Application.FormUrlEncoded
-        )
-
-        accessTokenInterceptor?.invoke(this)
-    }
+    val body = // some of really existing OAuth servers don't support other accept header values so keep it
+        client.post(baseUrl) {
+            header(HttpHeaders.Authorization, authHeader.render(HeaderValueEncoding.URI_ENCODE))
+            header(HttpHeaders.Accept, "*/*")
+            // some of really existing OAuth servers don't support other accept header values so keep it
+
+            setBody(
+                WriterContent(
+                    { params.formUrlEncodeTo(this) },
+                    ContentType.Application.FormUrlEncoded
+                )
+            )
+
+            accessTokenInterceptor?.invoke(this)
+        }.body<String>()
```

Comment:
Maybe we can face the same problem with `body<T>` as with `get<T>`. Should we also have some field without generic type?

Reply:
Added `bodyAsText` and `bodyAsChannel` extensions for visibility.